lib/elixir/lib/stream.ex

```diff
@@ -1404,6 +1401,22 @@ defmodule Stream do
     do: %{lazy | funs: [fun | funs], accs: [acc | accs], done: done}
   defp lazy(enum, acc, fun, done),
     do: %Stream{enum: enum, funs: [fun], accs: [acc], done: done}
+
+  defp do_interval({:first_time, interval}) do
```

Comment:
Maybe `interval_unfold/1` as the name? We should avoid `iterator` because it is not a word we use anywhere.